pkg/ccl/changefeedccl/sink_cloudstorage.go

```diff
@@ -648,21 +662,55 @@ func (s *cloudStorageSink) flushFile(ctx context.Context, file *cloudStorageSink
 		return file.flushToStorage(ctx, s.es, dest, s.metrics)
 	}
 
-	s.flushGroup.Add(1)
-	go func() {
-		defer s.flushGroup.Done()
-		// NB: must use s.flushCtx; ctx may be short lived (i.e. cancelled).
-		if err := file.flushToStorage(s.flushCtx, s.es, dest, s.metrics); err != nil {
-			log.Errorf(ctx, "error flushing file to storage: %s", err)
-			// We must use the same type for error we store in flushErr.
-			s.flushErr.CompareAndSwap(nil, &flushError{error: err})
+	s.flushQueue.Lock()
+	defer s.flushQueue.Unlock()
+	if s.flushQueue.q == nil {
```

Comment:
If we launch the goroutine here we don't append to the queue, so it's still nil if we were to come here again, and we'd launch yet another goroutine, which seems bad. It seems to me like we ought to just always append to the queue and then have `asyncFlush` not take arguments.

Reply:
Ack.  Goog point.